client/db.go

```diff
@@ -423,9 +426,9 @@ func sendAndFill(send func(...proto.Call) error, b *Batch) error {
 // Upon completion, Batch.Results will contain the results for each
 // operation. The order of the results matches the order the operations were
 // added to the batch.
-func (db *DB) Run(b *Batch) error {
+func (db *DB) Run(b *Batch) *proto.Error {
```

Comment:
That was the case before (see #2579), and (at least I) always found it a source of confusion to have an error wrapping another error (if you get an `error` and interface match it against errors you expect, you're going to miss it when it's wrapped). I feel that much clarity has been won by explicitly returning a `*proto.Error` here (and nothing's lost, since you get the thing you had before by just calling `GoError()`). We're moving a little bit away from idiomatic go to idiomatic use of the `req -> (resp, err)` pattern, but I think even on the `Go` side it's a net win (it's easy to have `*proto.Error` implement `error` again and then it'll be obvious which code paths relied on the wrapped error - the other way around wasn't so obvious at all).